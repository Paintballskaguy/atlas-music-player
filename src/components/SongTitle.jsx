import React from "react";

const SongTitle = ({ title, artist }) => {
  return (
    <div className="text-center">
      <h2 className="text-black text-xl font-bold">{title}</h2>
      <p className="text-gray-400 text-sm">{artist}</p>
    </div>
  );
};

export default SongTitle;