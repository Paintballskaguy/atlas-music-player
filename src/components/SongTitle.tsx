import React from "react";

interface SongTitleProps {
  title: string;
  artist: string;
}

const SongTitle: React.FC<SongTitleProps> = ({ title, artist }) => {
  return (
    <div className="text-center">
      <h2 className="text-custom-teal-800 dark:text-custom-teal-100 text-4xl font-black mb-3 tracking-tight leading-tight drop-shadow-sm">
        {title}
      </h2>
      <p className="text-custom-blue-600 dark:text-custom-blue-300 text-xl font-medium tracking-wide">
        {artist}
      </p>
    </div>
  );
};

export default SongTitle;