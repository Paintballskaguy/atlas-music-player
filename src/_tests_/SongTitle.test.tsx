import { describe, it, expect } from 'vitest'
import { render } from '@testing-library/react'
import SongTitle from '../components/SongTitle'

interface SongTitleProps {
  title: string
  artist?: string
  isPlaying?: boolean
  className?: string
}

describe('SongTitle', () => {
  it('renders with title and artist', () => {
    const props: SongTitleProps = {
      title: "Test Song",
      artist: "Test Artist"
    }
    
    const { container } = render(<SongTitle {...props} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders with only title', () => {
    const props: SongTitleProps = {
      title: "Test Song"
    }
    
    const { container } = render(<SongTitle {...props} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders with long title and artist', () => {
    const props: SongTitleProps = {
      title: "Very Long Song Title That Might Wrap", 
      artist: "Very Long Artist Name That Might Also Wrap"
    }
    
    const { container } = render(<SongTitle {...props} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders with custom className', () => {
    const props: SongTitleProps = {
      title: "Test Song",
      artist: "Test Artist",
      className: "custom-style"
    }
    
    const { container } = render(<SongTitle {...props} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders with isPlaying state', () => {
    const props: SongTitleProps = {
      title: "Test Song",
      artist: "Test Artist",
      isPlaying: true
    }
    
    const { container } = render(<SongTitle {...props} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})