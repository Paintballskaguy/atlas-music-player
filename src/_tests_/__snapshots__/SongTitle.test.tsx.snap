// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SongTitle > renders with custom className 1`] = `
<div
  class="text-center"
>
  <h2
    class="text-custom-teal-800 dark:text-custom-teal-100 text-4xl font-black mb-3 tracking-tight leading-tight drop-shadow-sm"
  >
    Test Song
  </h2>
  <p
    class="text-custom-blue-600 dark:text-custom-blue-300 text-xl font-medium tracking-wide"
  >
    Test Artist
  </p>
</div>
`;

exports[`SongTitle > renders with isPlaying state 1`] = `
<div
  class="text-center"
>
  <h2
    class="text-custom-teal-800 dark:text-custom-teal-100 text-4xl font-black mb-3 tracking-tight leading-tight drop-shadow-sm"
  >
    Test Song
  </h2>
  <p
    class="text-custom-blue-600 dark:text-custom-blue-300 text-xl font-medium tracking-wide"
  >
    Test Artist
  </p>
</div>
`;

exports[`SongTitle > renders with long title and artist 1`] = `
<div
  class="text-center"
>
  <h2
    class="text-custom-teal-800 dark:text-custom-teal-100 text-4xl font-black mb-3 tracking-tight leading-tight drop-shadow-sm"
  >
    Very Long Song Title That Might Wrap
  </h2>
  <p
    class="text-custom-blue-600 dark:text-custom-blue-300 text-xl font-medium tracking-wide"
  >
    Very Long Artist Name That Might Also Wrap
  </p>
</div>
`;

exports[`SongTitle > renders with only title 1`] = `
<div
  class="text-center"
>
  <h2
    class="text-custom-teal-800 dark:text-custom-teal-100 text-4xl font-black mb-3 tracking-tight leading-tight drop-shadow-sm"
  >
    Test Song
  </h2>
  <p
    class="text-custom-blue-600 dark:text-custom-blue-300 text-xl font-medium tracking-wide"
  />
</div>
`;

exports[`SongTitle > renders with title and artist 1`] = `
<div
  class="text-center"
>
  <h2
    class="text-custom-teal-800 dark:text-custom-teal-100 text-4xl font-black mb-3 tracking-tight leading-tight drop-shadow-sm"
  >
    Test Song
  </h2>
  <p
    class="text-custom-blue-600 dark:text-custom-blue-300 text-xl font-medium tracking-wide"
  >
    Test Artist
  </p>
</div>
`;
